### DISCUSSION TITLE
Feature Request: Ability to link with other ActivityPub accounts

### DISCUSSION THREAD

**Discussion by LivInTheLookingGlass** [mastodon/mastodon#16604] [General]
Basically the problem is this: in an ideal world, I would have one account that I can log in with at both my Mastodon instance of choice and a more specialized platform like BookWyrm. Maybe in the backend they are separate, but to an end user, I would like them to look the same.

Unfortunately, most services don't offer that. In order to emulate this behavior, it would be neat if we had an official way to "link" our accounts together. This would basically mean that you pick one profile to be your main identity, and you tell it in the settings to automatically boost anything posted by the other identities. Additionally, anything that would trigger a notification at the other identity (possibly excepting direct messages) would also trigger a notification on the main identity.

In order to prevent spam and abuse, it should be required that the enabling user post a status on the remote identity. This status will have some nonce given by the server to verify control over the account. Another way of confirming control would be through mutual keybase verification.

Is this a feature that others would be interested? Is this something the developers think is feasible or would be willing to allow?

**Comment by weex** [comment]
Restating your problem: You don't want more than one account across the Fediverse but you also don't want authentication to depend on only one site or instance.

**Reply by LivInTheLookingGlass** [reply]
Basically, yes. I realize that's a hairy problem, but my hope is that this is enough to not need a more sweeping solution

**Reply by weex** [reply]
If autoboosting was done in such a way that it didn't create new posts, but just decorated and passed on existing ones, I think that would be neat.

For the auth piece, Oauth was made for this but I wonder if anyone's ever created a service that allowed registration of more than one remote auth provider. For example you could login to two different mastodon instances and then for other accounts on say PixelFed, you could oauth with one, then in that session oauth to the other to establish one as a backup.

**Comment by roke-julian-lockhart** [comment]
This is already possible with aliases.

**Comment by trwnh** [comment]
See the following issues:

- #9206
- #19924
- #21413
